site/static/scripts/ui/domInteraction.js

```diff
@@ -0,0 +1,356 @@
+
+            populateDeviceActivators(dev.name, dev.deviceId, dev.activators);
+        };
+
+        elem.appendChild(document.createTextNode(device.name));
+
+        var renameButton = document.createElement("BUTTON");
+        renameButton.innerText = "Rename";
+        renameButton.onclick = submitRenameDevice;
+        elem.appendChild(renameButton);
+
+
+        namesListElem.appendChild(elem);
+    });
+
+    if (SELECTED_DEVICE == null){
+        if (namesListElem.firstChild != null){
+            namesListElem.firstChild.click();
+        }
+    } else {
+        document.getElementById(SELECTED_DEVICE).click();
+    }
+}
+
+/**
+ * Generates the html to view a device with a particular deviceName and for its
+ * associated activators. deviceId is passed so that we may interact with the
+ * activators and know their device. This will be refactored.
+ * @param {} deviceName
+ * @param {} deviceId
+ * @param {} activators
+ */
+function populateDeviceActivators(deviceName, deviceId, activators) {
+    var activatorsElem = document.getElementById("activatorsList");
+    var activatorsTitle = document.getElementById("activatorsTitle");
+    //Sort the activators alphabetically.
+    activators.sort(function(a, b){
+    	var nameA = a.name.toLowerCase(), nameB = b.name.toLowerCase();
+    	if (nameA < nameB) return -1;
+    	if (nameA > nameB) return 1;
+    	return 0;
+    })
+    activatorsTitle.firstChild.innerHTML = deviceName;
+    removeChildren(activatorsElem);
+    for (var i = 0; i < activators.length; i++) {
+        var activator = activators[i];
+        var elem = document.createElement("div");
+        elem.className = "device_control_row";
+        elem.appendChild(document.createTextNode(activator.name));
+        switch (activator.type) {
+            case "bool":
+                var label = document.createElement("label");
+                label.className = "switch";
+                var input = document.createElement("input");
+                input.type = "checkbox";
+                input.name = deviceId;
+                input.onclick = onToggleInteracted;
+                input.id = activator.activatorId;
+                input.checked = activator.state;
+                label.appendChild(input);
+                var span = document.createElement("span");
+                span.className = "switchSlider round";
+                label.append(span);
+                elem.appendChild(label);
+                break;
+            case "float":
+                var slideContainer = document.createElement("div");
+                slideContainer.className = "slidecontainer";
+                var input = document.createElement("input");
+                input.type = "range";
+                input.min = 0;
+                input.max = 100;
+                input.step = 10;
+                input.name = deviceId;
+                input.value = activator.state * 100; // TODO Based on object
```

Comment:
TODOs in the release isn't really appropriate.

Reply:
🤦‍♂️ 👍 We will leave To-Do out of the code for now on.